### PR TITLE
is payable by sc implementation on vm-common

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

File: interface.go [hunk 1]
```diff
@@ -198,7 +198,7 @@ type ESDTRoleHandler interface {
 
 // PayableHandler provides IsPayable function which returns if an account is payable or not
 type PayableHandler interface {
-	IsPayable(address []byte) (bool, error)
+	IsPayable(sndAddress, rcvAddress []byte) (bool, error)
```
Comment (iulianpascalau) [inline interface.go:201]: Not a big fan of having consecutive arguments of the same type as someone can easily switch between them.
Using arguments here, in the interface definition will clutter things up. Can remain as it is.

File: codeMetadata.go [hunk 2]
```diff
@@ -9,11 +9,14 @@ const (
 	MetadataPayable = 2
 	// MetadataReadable is the bit for readable flag
 	MetadataReadable = 4
+	// MetadataPayableBySC is the bit for payable flag
+	MetadataPayableBySC = 4
```
Comment (iulianpascalau) [inline codeMetadata.go:13]: This is confusing, we have to, at least, split them in 2 categories: the ones applied on the first byte and the ones applied on the second byte and so on.
At the first read I thought you just reused the value from the `MetadataReadable` . After further reading the whole code, I've found out that the `MetadataReadable` was applied on the first byte, not the second

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: interface.go (hunk 1 above)
Comment (sasurobert) [reply interface.go:201]: Yes. Will leave as it is.

File: codeMetadata.go (hunk 2 above)
Comment (sasurobert) [reply codeMetadata.go:13]: ok. split in too const groups

**Review by iulianpascalau — DISMISSED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by camilbancioiu — APPROVED** [review]

**Pull request merged**